infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -419,114 +495,141 @@ public void resetServer(CorfuPayloadMsg<SequencerRecoveryMsg> msg,
-            for(Map.Entry<UUID, StreamAddressSpace> streamAddressSpace : addressSpaceMap.entrySet()) {
+            for (Map.Entry<UUID, StreamAddressSpace> streamAddressSpace :
+                    addressSpaceMap.entrySet()) {
                 Long streamTail = streamAddressSpace.getValue().getTail();
-                log.trace("On Sequencer reset, tail for stream {} set to {}", streamAddressSpace.getKey(), streamTail);
+                if (log.isTraceEnabled()) {
+                    log.trace("On Sequencer reset, tail for stream {} set to {}",
+                            streamAddressSpace.getKey(), streamTail);
+                }
                 streamTailToGlobalTailMap.put(streamAddressSpace.getKey(), streamTail);
             }
 
             // Reset streams address map
-            this.streamsAddressMap = new HashMap<>();
-            this.streamsAddressMap.putAll(addressSpaceMap);
+            streamsAddressMap = new HashMap<>();
+            streamsAddressMap.putAll(addressSpaceMap);
 
-            for (Map.Entry<UUID, StreamAddressSpace> streamAddressSpace : this.streamsAddressMap.entrySet()) {
-                log.info("Stream[{}] set to last trimmed address {} and {} addresses in the range [{}-{}], " +
-                                "on sequencer reset.",
+            for (Map.Entry<UUID, StreamAddressSpace> streamAddressSpace :
+                    streamsAddressMap.entrySet()) {
+                log.info("Stream[{}] set to last trimmed address {} and {} addresses in the range" +
+                                " [{}-{}], on sequencer reset.",
                         Utils.toReadableId(streamAddressSpace.getKey()),
                         streamAddressSpace.getValue().getTrimMark(),
                         streamAddressSpace.getValue().getAddressMap().getLongCardinality(),
                         streamAddressSpace.getValue().getLowestAddress(),
                         streamAddressSpace.getValue().getHighestAddress());
                 if (log.isTraceEnabled()) {
                     log.trace("Stream[{}] address map on sequencer reset: {}",
-                            Utils.toReadableId(streamAddressSpace.getKey()), streamAddressSpace.getValue().getAddressMap());
+                            Utils.toReadableId(streamAddressSpace.getKey()),
+                            streamAddressSpace.getValue().getAddressMap());
                 }
             }
         }
 
         // Update epochRangeLowerBound if the bootstrap epoch is not consecutive.
-        if (epochRangeLowerBound == Layout.INVALID_EPOCH || bootstrapMsgEpoch != sequencerEpoch + 1) {
+        if (epochRangeLowerBound == Layout.INVALID_EPOCH
+                || bootstrapMsgEpoch != sequencerEpoch + 1) {
             epochRangeLowerBound = bootstrapMsgEpoch;
         }
 
         // Mark the sequencer as ready after the tails have been populated.
-        sequencerEpoch = bootstrapMsgEpoch;
+        setSequencerEpoch(bootstrapMsgEpoch);
```

Comment:
sequencerEpoch should probably be a private volatile.

Reply:
Done,  thanks!